src/libstd/sys/common/thread_info.rs

```diff
@@ -57,6 +57,10 @@ pub fn set(stack_guard: Option<usize>, thread: Thread) {
 }
 
 // a hack to get around privacy restrictions; implemented by `std::thread`
+#[doc(hidden)]
+#[unstable(feature = "libstd_sys_internals",
+           reason = "this trait is considered exported for some reason",
```

Comment:
You probably meant this issue https://github.com/alexcrichton/rust/commit/0daaeab244a583b6bb5b88c41ad0fdd51388619d